### PR TITLE
[Core] Config not required for pipeline client

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @bryevdv!
Hmm well this is kind of cleaning up the code that's there - we allow users to pass in policies or a pipeline, and therefore there would be no need to pass in a config.
I think for storage we build the pipeline ourselves and pass it in. In the case of cosmos we don't use any kind of configuration object, so pass in a list of policies. My understanding is that the config object itself is being phased out - it's already been removed from public-facing APIs.

File: sdk/core/azure-core/azure/core/pipeline_client.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ class PipelineClient(PipelineClientBase):
     Builds a Pipeline client.
 
     :param str base_url: URL for the request.
-    :param config: Service configuration. This is a required parameter.
+    :param config: Service configuration. This is a required unless pipeline or policies are given separately.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline_client.py]: Missing "parameter"?

File: sdk/core/azure-core/azure/core/pipeline_client.py [hunk 2]
```diff
@@ -80,10 +80,8 @@ class PipelineClient(PipelineClientBase):
             :caption: Builds the pipeline client.
     """
 
-    def __init__(self, base_url, config, **kwargs):
+    def __init__(self, base_url, config=None, **kwargs):
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline_client.py]: Should we put config into kwargs?

File: sdk/core/azure-core/azure/core/pipeline_client.py [hunk 3]
```diff
@@ -114,8 +114,8 @@ def _build_pipeline(self, config, **kwargs): # pylint: disable=no-self-use
                 config.redirect_policy,
                 config.retry_policy,
                 config.custom_hook_policy,
-                config.logging_policy,
                 DistributedTracingPolicy(),
+                config.logging_policy,
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline_client.py]: Maybe we should add proxy policy as well.

File: sdk/core/azure-core/azure/core/pipeline_client.py [hunk 4]
```diff
@@ -106,6 +104,8 @@ def _build_pipeline(self, config, **kwargs): # pylint: disable=no-self-use
         policies = kwargs.get('policies')
 
         if policies is None:  # [] is a valid policy list
+            if config is None:
+                raise ValueError("No config or policies supplied.")
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline_client.py]: Why not if config is None, give a default one?

**Comment by annatisch** [comment]
This PR isn't needed for Preview3 - so going to shelve for now.

**Comment by annatisch** [comment]
Picked up in PR #7734

**Pull request closed without merging**